aws/table_aws_shield_emergency_contact.go

```diff
@@ -0,0 +1,96 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/shield"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsShieldEmergencyContact(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_shield_emergency_contact",
+		Description: "AWS Shield Emergency Contact",
+		List: &plugin.ListConfig{
+			Hydrate: listAwsShieldEmergencyContact,
+			Tags:    map[string]string{"service": "shield", "action": "DescribeEmergencyContactSettings"},
+		},
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "priority",
+				Description: "The priority of the contact in the emergency contact list.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("Priority"),
+			},
+			{
+				Name:        "email_address",
+				Description: "The email address for the contact.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("EmailAddress"),
+			},
+			{
+				Name:        "phone_number",
+				Description: "The phone number for the contact.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("PhoneNumber"),
+			},
+			{
+				Name:        "ContactNotes",
+				Description: "Additional notes regarding the contact.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ContactNotes"),
+			},
+		}),
+	}
+}
+
+type emergencyContact struct {
+	Priority int
+	EmailAddress *string
+    PhoneNumber *string
+	ContactNotes *string
+}
+
+//// HYDRATE FUNCTIONS
+
+func listAwsShieldEmergencyContact(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := ShieldClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_shield_emergency_contact.listAwsShieldEmergencyContact", "connection_error", err)
+		return nil, err
+	}
+
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	data, err := svc.DescribeEmergencyContactSettings(ctx, &shield.DescribeEmergencyContactSettingsInput{})
+
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_shield_emergency_contact.listAwsShieldEmergencyContact", "api_error", err)
+		return nil, err
+	}
+
+	for i, contact := range data.EmergencyContactList {
+		d.StreamListItem(ctx, &emergencyContact{
+			Priority:		i,
```

Comment:
I added this column to the table, since the order in which the emergency contacts are defined matters. The SRT will contact contacts based on the order in which they are defined.

Reply:
In my opinion, we shouldn't add extra columns beyond what the API returns. If the API provides the order of contact, then it makes sense to include that instead.

Reply:
I was also pondering about this, because the API returns the contacts as a list via the `EmergencyContactList` field. From my perspective it is more convenient for the user if the list is already unpacked for him, otherwise every query for the table would require a cross join.

But if this is violating the conventions how you usually handle these kind of API responses, then we can also have only one column in this table named `emergency_contact_list` which will contain the JSON array.